src/components/calcite-shell-panel/calcite-shell-panel.tsx

```diff
@@ -86,25 +142,222 @@ export class CalciteShellPanel {
+
+  setContentWidth(width: number): void {
+    const { contentWidthMax, contentWidthMin, position } = this;
+
+    const rounded = Math.round(position === "end" ? width : width);
+
+    this.contentWidth =
+      typeof contentWidthMax === "number" && typeof contentWidthMin === "number"
+        ? clamp(rounded, contentWidthMin, contentWidthMax)
+        : rounded;
+  }
+
+  updateAriaValues = (): void => {
+    const { contentEl } = this;
+    const computedStyle = contentEl && getComputedStyle(contentEl);
+
+    if (!computedStyle) {
+      return;
+    }
+
+    const max = parseInt(computedStyle.getPropertyValue("max-width"), 10);
+    const min = parseInt(computedStyle.getPropertyValue("min-width"), 10);
+    const valueNow = parseInt(computedStyle.getPropertyValue("width"), 10);
+
+    if (typeof valueNow === "number" && !isNaN(valueNow)) {
+      this.initialContentWidth = valueNow;
+    }
+
+    if (typeof max === "number" && !isNaN(max)) {
+      this.contentWidthMax = max;
+    }
+
+    if (typeof min === "number" && !isNaN(min)) {
+      this.contentWidthMin = min;
+    }
+
+    forceUpdate(this);
+  };
+
+  storeContentEl = (contentEl: HTMLDivElement): void => {
+    this.contentEl = contentEl;
+  };
+
+  getKeyvalue = (event: KeyboardEvent): number => {
+    const key = getKey(event.key);
+    const {
+      step,
+      stepMultiplier,
+      contentWidthMin,
+      contentWidthMax,
+      initialContentWidth,
+      position
+    } = this;
+    const multipliedStep = step * stepMultiplier;
+
+    const MOVEMENT_KEYS = [
+      "ArrowUp",
+      "ArrowDown",
+      "ArrowLeft",
+      "ArrowRight",
+      "Home",
+      "End",
+      "PageUp",
+      "PageDown"
+    ];
+
+    if (MOVEMENT_KEYS.indexOf(key) > -1) {
+      event.preventDefault();
+      event.stopPropagation();
+    }
+
+    const increaseKeys =
+      key === "ArrowUp" || (position === "end" ? key === "ArrowLeft" : key === "ArrowRight");
+
+    if (increaseKeys) {
+      const stepValue = event.shiftKey ? multipliedStep : step;
+
+      return initialContentWidth + stepValue;
+    }
+
+    const decreaseKeys =
+      key === "ArrowDown" || (position === "end" ? key === "ArrowRight" : key === "ArrowLeft");
+
+    if (decreaseKeys) {
+      const stepValue = event.shiftKey ? multipliedStep : step;
+
+      return initialContentWidth - stepValue;
+    }
+
+    if (typeof contentWidthMin === "number" && key === "Home") {
+      return contentWidthMin;
+    }
+
+    if (typeof contentWidthMax === "number" && key === "End") {
+      return contentWidthMax;
+    }
+
+    if (key === "PageDown") {
+      return initialContentWidth - multipliedStep;
+    }
+
+    if (key === "PageUp") {
+      return initialContentWidth + multipliedStep;
+    }
+
+    return null;
+  };
+
+  separatorKeyDown = (event: KeyboardEvent): void => {
+    this.setInitialContentWidth();
+    const width = this.getKeyvalue(event);
```

Comment:
Nitpick: Can you rename this to better express its intent? Something like `getKeyAdjustedWidth` or along those lines? 

On my first scan I thought this method returned a value based on the key.